src/NuGet.Core/NuGet.Logging/Logger.cs

```diff
@@ -0,0 +1,22 @@
+﻿// Copyright (c) .NET Foundation. All rights reserved.
+// Licensed under the Apache License, Version 2.0. See License.txt in the project root for license information.
+
+namespace NuGet.Logging
+{
+    public class Logger
+    {
+        private static ILogger _logger = new NullLogger();
+
+        public static ILogger Instance
+        {
+            get
+            {
+                return _logger;
+            }
+            set
+            {
+                _logger = value ?? new NullLogger();
```

Comment:
This pattern is to avoid nullrefs anywhere code tries to log something using `Logger.Instance`. DNX uses a similar pattern: https://github.com/aspnet/dnx/blob/dev/src/Microsoft.Dnx.Tooling/Reports.cs#L15

Happy to learn about any alternative suggestions though.